ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -87,17 +96,17 @@ export default class PermissionConnect extends Component {
 
   selectAccount = (address) => {
     this.setState({
-      page: 2,
       selectedAccountAddress: address,
     })
+    this.props.history.push(this.props.confirmPermissionPath)
```

Comment:
This should be in the `setState` callback, to ensure it happens _after_ the state is set. Otherwise the next page might be rendered without a selected account.